internal/compiler/generator/rust.rs

```diff
@@ -1537,8 +1537,16 @@ fn generate_item_tree(
     };
 
     let parent_item_expression = parent_ctx.and_then(|parent| {
-        parent.repeater_index.map(|idx| {
-            let sub_component_offset = parent.ctx.current_sub_component().unwrap().repeated[idx].index_in_tree;
+        let current_sub_component = parent.ctx.current_sub_component().unwrap();
+        Some(parent.repeater_index.map_or_else(|| {
+            // No repeater index, this could be a PopupWindow
+            let window_tokens = access_window_adapter_field(parent.ctx);
+            quote!(let popup_item_tree = &sp::VRcMapped::origin(&self.as_ref().self_weak.get().unwrap().upgrade().unwrap());
+                   let _self = &self;
+                   sp::WindowInner::from_pub(#window_tokens.window()).get_popup_parent(popup_item_tree, _result);
```

Comment:
I think you don't need get_popup_parent.
You can do something like 
`sp::ItemRc::new(self.parent.upgrade(), popup_parent_index)`
the hard part is to figure out the popup_parent_index.  I guess for the purpose of finding the window, `0` would work, but ideally we'd store that in the parent_ctx

and then something similar can be done for C++

Reply:
Wow, this works. I don't really understand the bit about popup_parent_index though, isn't that always 0 to get to the main window? I guess in case of nested popups it means with 0 I'm jumping straight to the toplevel, which is incorrect for parent_node() in general, but ok for the default_font_size use case?

Reply:
To do this properly I would add
`pub popup_index : SubComponentIdx;`
in `ParentCtx` ? But then I'd need another test in order to figure out that I'm setting this correctly, given that my current testcase works without this.